src/main/webapp/app/faq/faq-update.component.ts

```diff
@@ -0,0 +1,153 @@
+import { ArtemisCategorySelectorModule } from 'app/shared/category-selector/category-selector.module';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+
+@Component({
+    selector: 'jhi-faq-update',
+    templateUrl: './faq-update.component.html',
+    styleUrls: ['./faq-update.component.scss'],
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisSharedComponentModule, ArtemisMarkdownEditorModule, ArtemisCategorySelectorModule],
+})
+export class FAQUpdateComponent implements OnInit {
+    faq: FAQ;
+    isSaving: boolean;
+    existingCategories: FAQCategory[];
+    faqCategories: FAQCategory[];
+
+    domainActionsDescription = [new FormulaAction()];
+
+    // Icons
+    faQuestionCircle = faQuestionCircle;
+    faSave = faSave;
+    faBan = faBan;
+
+    constructor(
+        protected alertService: AlertService,
+        protected faqService: FAQService,
+        protected activatedRoute: ActivatedRoute,
+        private navigationUtilService: ArtemisNavigationUtilService,
+        private router: Router,
+        private translateService: TranslateService,
+    ) {}
+
+    /**
+     * Life cycle hook called by Angular to indicate that Angular is done creating the component
+     */
+    ngOnInit() {
+        this.isSaving = false;
+        this.activatedRoute.parent?.data.subscribe((data) => {
+            // Create a new faq to use unless we fetch an existing faq
+            const faq = data['faq'];
+            this.faq = faq ?? new FAQ();
+            const course = data['course'];
+            if (course) {
+                this.faq.course = course;
+                this.loadCourseFaqCategories(course.id);
+            }
+            this.faqCategories = faq?.categories ? faq.categories : [];
+        });
+    }
+
+    /**
+     * Revert to the previous state, equivalent with pressing the back button on your browser
+     * Returns to the detail page if there is no previous state and we edited an existing faq
+     * Returns to the overview page if there is no previous state and we created a new faq
+     */
+
+    previousState() {
+        this.navigationUtilService.navigateBack(['course-management', this.faq.course!.id!.toString(), 'faqs']);
+    }
+    /**
+     * Save the changes on a faq
+     * This function is called by pressing save after creating or editing a faq
+     */
+    save() {
+        this.isSaving = true;
+        if (this.faq.id !== undefined) {
+            this.subscribeToSaveResponse(this.faqService.update(this.faq));
+        } else {
+            this.faq.faqState = FAQState.ACCEPTED;
+            this.subscribeToSaveResponse(this.faqService.create(this.faq));
+        }
+    }
+
+    /**
+     * @param result The Http response from the server
+     */
+    protected subscribeToSaveResponse(result: Observable<HttpResponse<FAQ>>) {
```

Comment:
Two questions:
1. Why `protected`-access. Do you plan to inherit this class?
2. This method serves two purposes - handling the result of update and creation whose execution you determine with an if-else, also beforehand. Wouldn't it be more understandable if you have two different methods for that?